common/contracts/split-warehouse.go

```diff
@@ -0,0 +1,79 @@
+)
+
+// ABI cache
+var splitWarehouseAbi abi.ABI
+var splitWarehouseOnce sync.Once
+
+type SplitWarehouse struct {
+	contract *eth.Contract
+	txMgr    *eth.TransactionManager
+}
+
+// Create a new SplitWarehouse instance
+func NewSplitWarehouse(address common.Address, ec eth.IExecutionClient, txMgr *eth.TransactionManager) (*SplitWarehouse, error) {
+	// Parse the ABI
+	var err error
+	splitWarehouseOnce.Do(func() {
+		var parsedAbi abi.ABI
+		parsedAbi, err = abi.JSON(strings.NewReader(splitWarehouseAbiString))
+		if err == nil {
+			splitWarehouseAbi = parsedAbi
+		}
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error parsing SplitWarehouse ABI: %w", err)
+	}
+
+	// Create the contract
+	contract := &eth.Contract{
+		ContractImpl: bind.NewBoundContract(address, splitWarehouseAbi, ec, ec, ec),
+		Address:      address,
+		ABI:          &splitWarehouseAbi,
+	}
+
+	return &SplitWarehouse{
+		contract: contract,
+		txMgr:    txMgr,
+	}, nil
+}
+
+// =============
+// === Calls ===
+// =============
+
+func (c *SplitWarehouse) BalanceOf(mc *batch.MultiCaller, out **big.Int, account common.Address, token common.Address) {
+	eth.AddCallToMulticaller(mc, c.contract, out, "balanceOf", account, ToUint256(token))
+}
+
+func (c *SplitWarehouse) NativeToken(mc *batch.MultiCaller, out *common.Address) {
+	eth.AddCallToMulticaller(mc, c.contract, out, "NATIVE_TOKEN")
+}
+
+// ====================
+// === Transactions ===
+// ====================
+
+func (c *SplitWarehouse) Withdraw(address common.Address, amountWithdraw []*big.Int, claimTokenList []common.Address, opts *bind.TransactOpts) (*eth.TransactionInfo, error) {
```

Comment:
Minor but I would suggest swapping the positions of the `amountWithdraw` and `claimTokenList` args here - makes more sense to list what to claim first then how much of it to claim after, and it has parity with the contract function.

Reply:
Fixed